ncm-openstack/src/main/pan/components/openstack/horizon.pan

```diff
@@ -0,0 +1,313 @@
+# ${license-info}
+# ${developer-info}
+# ${author-info}
+
+
+declaration template components/openstack/horizon;
+
+@documentation {
+    The Horizon configuration options in "caches" Section.
+}
+type openstack_horizon_caches = {
+     @{We recommend you use memcached for development; otherwise after every reload
+     of the django development server, you will have to login again}
+    'backend' : string = 'django.core.cache.backends.memcached.MemcachedCache'
+    @{location format <fqdn>:<port>}
+    'location' : type_hostport
+} = dict();
+
+@documentation {
+    The Horizon api versions section.
+    Overrides for OpenStack API versions. Use this setting to force the
+    OpenStack dashboard to use a specific API version for a given service API.
+    Versions specified here should be integers or floats, not strings.
+    NOTE: The version should be formatted as it appears in the URL for the
+    service API. For example, The identity service APIs have inconsistent
+    use of the decimal point, so valid options would be 2.0 or 3.
+    Minimum compute version to get the instance locked status is 2.9.
+}
+type openstack_horizon_api_versions = {
+    'identity' : long(1..) = 3
```

Comment:
this does not support microversions. you should introduce aversion type (and add it to https://github.com/quattor/template-library-core/blob/571476a475924ef1ad0f0bd8f65241d2adc48d3f/pan/types.pan)